well_known.go

```diff
@@ -16,7 +15,7 @@ type WellKnownResult struct {
 
 // LookupWellKnown looks up a well-known record for a matrix server. If one if
 // found, it returns the server to redirect to.
-func LookupWellKnown(serverNameType ServerName) (*WellKnownResult, error) {
+func LookupWellKnown(serverNameType ServerName) (*WellKnownResult, bool, error) {
```

Comment:
Passing though: Why a new boolean instead of `null, null`. A null error and a null WellKnownResult would imply that no well known was found, but it's also not an error?

Reply:
fwiw I did it that way because imho it makes the code easier to understand, but maybe the same thing could be achieved through returning `nil, nil` and documenting this behaviour in the doc comment on top of the function? Not sure which one is best here.

Reply:
It's personal taste, and I understand the reason. I think I'd put the boolean before the WellKnownResult but that might not be how Go prefers it. Feel free to ignore me :)